pkg/operator/api/api.go

```diff
@@ -74,6 +99,43 @@ func (c *Cluster) Start() error {
 	return nil
 }
 
+func (c *Cluster) makeClusterRole() error {
+	account := &v1.ServiceAccount{}
+	account.Name = DeploymentName
+	account.Namespace = c.Namespace
+	_, err := c.context.Clientset.CoreV1().ServiceAccounts(c.Namespace).Create(account)
+	if err != nil && !errors.IsAlreadyExists(err) {
+		return fmt.Errorf("failed to create api service account. %+v", err)
+	}
+
+	// Create the cluster role if it doesn't yet exist.
+	// If the role already exists we have to update it. Otherwise if the permissions change during an upgrade,
+	// the create will fail with an error that we're changing the permissions.
+	role := &v1beta1.ClusterRole{Rules: clusterAccessRules}
+	role.Name = DeploymentName
+	_, err = c.context.Clientset.RbacV1beta1().ClusterRoles().Get(role.Name, metav1.GetOptions{})
+	if errors.IsNotFound(err) {
+		logger.Infof("creating cluster role rook-api")
+		_, err = c.context.Clientset.RbacV1beta1().ClusterRoles().Create(role)
+	} else if err == nil {
+		logger.Infof("cluster role rook-api already exists. updating if needed.")
+		_, err = c.context.Clientset.RbacV1beta1().ClusterRoles().Update(role)
+	}
+	if err != nil {
+		return fmt.Errorf("failed to create cluster roles. %+v", err)
+	}
+
+	binding := &v1beta1.ClusterRoleBinding{}
+	binding.Name = DeploymentName
+	binding.RoleRef = v1beta1.RoleRef{Name: DeploymentName, Kind: "ClusterRole", APIGroup: "rbac.authorization.k8s.io"}
```

Comment:
i would have thought that the role object created above would be used to create or populate the `RoleRef`.  But I guess that's not necessary since the only field they would share would be `Name`?